File: exercises/week_01/cpp/student.cc
#include "student.h"

#include <iostream>

Student::Student() {
  this->set_credits(0);
}

Student::Student(std::string name, std::string student_number) {
  this->set_name(name);
  this->set_student_number(student_number);
  this->set_credits(0);
}

std::string Student::name() { return this->name_; }
void Student::set_name(std::string name) {
  this->name_ = name;
}

std::string Student::student_number() { return this->student_number_; }
void Student::set_student_number(std::string student_number) {
  this->student_number_ = student_number;
}

int Student::credits() { return this->credits_; }
void Student::set_credits(int credits) {
  this->credits_ = credits;
}

void Student::Print() {
  std::cout << "{ "
            << "\"name\": "           << "\"" << this->name() << "\", "
            << "\"student_number\": " << "\"" << this->student_number() << "\", "
            << "\"credits\": "        << this->credits()
            << " }\n";
}

File: exercises/week_01/java/Student.java
public class Student {
    private String name;
    private String studentNumber;
    private int credits;

    public Student() { }

    public Student(String name, String studentNumber) {
        this.name = name;
        this.studentNumber = studentNumber;
    }

    public String getName() { return this.name; }
    public void setName(String name) { this.name = name; }

    public String getStudentNumber() { return this.studentNumber; }
    public void setStudentNumber(String studentNumber) { this.studentNumber = studentNumber; }

    public int getCredits() { return this.credits; }
    public void setCredits(int credits) { this.credits = credits; }

    public void Print() {
        var format = "{ \"name\": \"%s\", \"student_number\":, \"%s\", \"credits\": %d }";
        
        var json = String.format(
            format, 
            this.getName(),
            this.getStudentNumber(),
            this.getCredits());

        System.out.println(json);
    }
}
